Sources/SwiftDocC/Model/Rendering/RenderSectionTranslator/DeclarationsSectionTranslator.swift

```diff
@@ -28,39 +29,61 @@ struct DeclarationsSectionTranslator: RenderSectionTranslator {
             for pair in declaration {
                 let (platforms, declaration) = pair
                 
-                let renderedTokens = declaration.declarationFragments.map { token -> DeclarationRenderSection.Token in
-                    
-                    // Create a reference if one found
-                    var reference: ResolvedTopicReference?
-                    if let preciseIdentifier = token.preciseIdentifier,
-                       let resolved = renderNodeTranslator.context.symbolIndex[preciseIdentifier] {
-                        reference = resolved
-                        
-                        // Add relationship to render references
-                        renderNodeTranslator.collectedTopicReferences.append(resolved)
-                    }
-                    
-                    // Add the declaration token
-                    return DeclarationRenderSection.Token(fragment: token, identifier: reference?.absoluteString)
-                }
-                
                 let platformNames = platforms.sorted { (lhs, rhs) -> Bool in
                     guard let lhsValue = lhs, let rhsValue = rhs else {
                         return lhs == nil
                     }
                     return lhsValue.rawValue < rhsValue.rawValue
                 }
                 
+                let renderedTokens = renderDeclarationTokens(fragments: declaration.declarationFragments, 
+                                                             renderNodeTranslator: &renderNodeTranslator)
+
+                // If this symbol has overloads, render their declarations as well.
+                let overloadsVariant = symbol.overloadsVariants[trait]
+                // TODO: Support overloads that have platform variants
+                let otherDeclarations = overloadsVariant?.compactMap { overloadReference -> DeclarationRenderSection.OtherDeclaration? in
+                    guard let overload = try? renderNodeTranslator.context.entity(with: overloadReference).semantic as? Symbol,
+                          let declarationFragments = overload.declarationVariants[trait]?.values.first?.declarationFragments else {
+                        return nil
```

Comment:
Should we assert here? What does it mean if we get an overload that is not a symbol here, or that does not have any declaration fragments? Is that a valid scenario? Are there any symbols with no declaration at all?

Silently skipping over invalid symbols might make bugs elsewhere in the code harder to find.

Reply:
I have a check back in the DocumentationContext that each overload must be a symbol, do you think I should assert here as well? 
I've added an assertion that all overloads must have declaration fragments.